source/base/function_signed_distance.cc

```diff
@@ -126,6 +128,126 @@ namespace Functions
       return SymmetricTensor<2, dim>();
     }
 
+
+
+    template <int dim>
+    Ellipsoid<dim>::Ellipsoid(const Point<dim> &             center,
+                              const std::array<double, dim> &radii,
+                              const double                   tolerance,
+                              const unsigned int             max_iter)
+      : center(center)
+      , radii(radii)
+      , tolerance(tolerance)
+      , max_iter(max_iter)
+    {
+      for (int d = 0; d < dim; ++d)
+        Assert(radii[d] > 0, ExcMessage("All radii must be positive."))
+    }
+
+
+
+    template <int dim>
+    double
+    Ellipsoid<dim>::value(const Point<dim> & point,
+                          const unsigned int component) const
+    {
+      AssertIndexRange(component, this->n_components);
```

Comment:
The test suite is complaining. This should fix it:
```suggestion
      AssertIndexRange(component, this->n_components);
      (void)component;
```